### PR TITLE
docs: Add Getting Started docs for clustermesh service affinity

### PR DESCRIPTION
**Pull request by sayboras**
This commit is to add getting started docs for global service with
affinity feature.

Relates: #19521
Signed-off-by: Tam Mach <tam.mach@cilium.io>

### CODE REVIEW

File: Documentation/gettingstarted/clustermesh/services.rst [hunk 1]
```diff
@@ -110,3 +142,61 @@ Deploying a Simple Example Service
       kubectl exec -ti deployment/x-wing -- curl rebel-base
 
    You will see replies from pods in both clusters again.
+
+9. In cluster 1, add ``io.cilium/service-affinity="local"`` to existing global service
```
Comment (joestringer) [inline Documentation/gettingstarted/clustermesh/services.rst]: I think this should be a separate section? This makes it look like it requires 16 steps just to get a global service running :thinking:

File: Documentation/gettingstarted/clustermesh/services.rst [hunk 2]
```diff
@@ -110,3 +142,61 @@ Deploying a Simple Example Service
...
+
+   .. code-block:: shell-session
+
+      kubectl annotate service rebel-base io.cilium/service-affinity="local" --overwrite
+
+10. From cluster 1, access the global service:
```
Comment (joestringer) [inline Documentation/gettingstarted/clustermesh/services.rst]: nit: If each of these numbers is `1.` and the content inside each of these numbered bullets is indented starting from the same indentation as the non-space character following the `1.`, then sphinx should autonumber these for you. :sweat_smile: 

(But I'm also okay with just writing all the numbers out like this if it's easier).

**Review by joestringer — CHANGES_REQUESTED** [review]
Goods & Services Tax? :sweat_smile: 

Given that the current example is "Deploying a Simple Example Service", I wonder if we should put the new instructions into a new section afterwards that highlights this different feature. That can help people understand that you don't have to go through so many steps to configure a global service, but also that this more advanced feature can be configured and demonstrated in a similar short number of steps. What do you think?

File: Documentation/gettingstarted/clustermesh/services.rst (hunk 1 above)
Comment (sayboras) [reply Documentation/gettingstarted/clustermesh/services.rst]: good point, let me split it out 💯 

Yeah, i just paid Good Service Tax yesterday, so just use the same term for Getting Started 🙈

File: Documentation/gettingstarted/clustermesh/services.rst (hunk 2 above)
Comment (sayboras) [reply Documentation/gettingstarted/clustermesh/services.rst]: good tip :100: , but I keep the running sequence to be consistent with other pages.

**Review by joestringer — APPROVED** [review]
I didn't try the steps at all, but LGTM from docs-structure.

File: Documentation/gettingstarted/clustermesh/affinity.rst [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+You need to have a functioning Cluster Mesh with a Global Service, please
+follow the guide :ref:`gs_clustermesh` and :ref:`gs_clustermesh_services`
+to set it up.
+
+Enabling Global Service Affinity
+################################
+
+Load-balancing across multiple clusters might not be ideal in some cases.
+The annotation ``io.cilium/service-affinity: "local|remote|none"`` can be used
+to specify the ``preferred`` endpoint destination.
```
Comment (jrajahalme) [inline Documentation/gettingstarted/clustermesh/affinity.rst]: Are the quotes around "preferred" needed, as it is not a keyword?

File: Documentation/gettingstarted/clustermesh/affinity.rst [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+For example, if the value of annotation ``io.cilium/service-affinity`` is local,
+the Global Service will load-balance across healthy local backends, and only user
```
Comment (jrajahalme) [inline Documentation/gettingstarted/clustermesh/affinity.rst]: Double-backtick word "local" here, as it is a keyword?

File: Documentation/gettingstarted/clustermesh/affinity.rst [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+     annotations:
+        io.cilium/global-service: "true"
+        # Possible values:
+        # - local
+        #    preferred endpoints from local cluster if available
+        # - remote
+        #    preferred endpoints from remote cluster if available
+        # none (default)
+        #    no preference. Default behavior if this annotation does not exist
+        "io.cilium/service-affinity": "local"
```
Comment (jrajahalme) [inline Documentation/gettingstarted/clustermesh/affinity.rst]: Inconsistent double quotes around the annotation names above.

File: Documentation/gettingstarted/clustermesh/affinity.rst [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+     - port: 80
+     selector:
+       name: rebel-base
+
+
+1. In cluster 1, add ``io.cilium/service-affinity="local"`` to existing global service
+
+   .. code-block:: shell-session
+
+      kubectl annotate service rebel-base io.cilium/service-affinity="local" --overwrite
```
Comment (jrajahalme) [inline Documentation/gettingstarted/clustermesh/affinity.rst]: double quotes not needed here.

File: Documentation/gettingstarted/clustermesh/affinity.rst [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+      5    10.96.173.113:80    ClusterIP      1 => 10.244.2.136:80 (active)
+                                              2 => 10.244.1.61:80 (active) (preferred)
+                                              3 => 10.244.2.31:80 (active) (preferred)
+                                              4 => 10.244.2.200:80 (active)
+
+5. In cluster 1, change ``io.cilium/service-affinity`` value to ``remote`` for existing global service
+
+   .. code-block:: shell-session
+
+      kubectl annotate service rebel-base io.cilium/service-affinity="remote" --overwrite
```
Comment (jrajahalme) [inline Documentation/gettingstarted/clustermesh/affinity.rst]: double quotes not needed here?

**Review by jrajahalme — APPROVED** [review]
Trivial comments on the formatting only.

File: Documentation/gettingstarted/clustermesh/affinity.rst (hunk 3 above)
Comment (sayboras) [reply Documentation/gettingstarted/clustermesh/affinity.rst]: good spot, it's not keyword, let me remove them.

**Pull request merged**